### DISCUSSION TITLE
assigning variables to cy.intercept() produces different behaviour to cy.server() with cy.route()

### DISCUSSION THREAD

**Discussion by matthew-nash-cko** [cypress-io/cypress#14259] [Component Testing]
Hi all,
Since swapping route for intercept i'm getting an issue whereby Cypress fails to spy on a route with a unique variable name, but with a path that has already been recorded/cached in the routes table.  example:

*** WORKING *** 
```js
Given('something', function() {
    cy.intercept("GET", "**/payments/*", {fixture:"path1"}).as("detailsPanel")
})

Given('something else', function() {
  **cy.server()
  cy.route("GET", "**/payments/*", "fixture:path2").as("newResponse")**
  cy.wait("@newResponse")
})
```

*** NOT WORKING ***

```js
Given('something', function() {
    cy.intercept("GET", "**/payments/*", {fixture:"path1"}).as("detailsPanel")
})

Given('something else', function() {
  **cy.intercept("GET", "**/payments/*", {fixture:"path2"}).as("newResponse")**
  cy.wait("@newResponse")
_// THIS CALL TIMES OUT BECAUSE THE ROUTE CAN'T BE FOUND_
})
```

Is anyone else experiencing this issue.  Apologies if this issue has been raised already

Matt

**Comment by jennifer-shehane** [comment]
It may be this issue, which will be fixed in our next release. https://github.com/cypress-io/cypress/issues/9379#issuecomment-742004697 I can't be sure without a clear way to reproduce.
